taurus/entity/dict_serializable.py

```diff
@@ -82,7 +82,40 @@ def build(d):
         return loads(dumps(d))
 
     def __repr__(self):
-        return str(self.as_dict())
+        object_dict = self.as_dict()
+        # as_dict() skips over keys in `skip`, but they should be in the representation.
+        skipped_keys = {x.lstrip('_') for x in vars(self) if x in self.skip}
+        for key in skipped_keys:
+            skipped_field = self.__getattribute__(key)
+            object_dict[key] = self._name_repr(skipped_field)
+        return object_dict.__str__()
+
+    def _name_repr(self, entity):
+        """
+        A representation of an object or a list of objects that uses the name and type.
+
+        This is used to represent soft-linked objects without inundating the user with
+        repetitive information.
+
+        Parameters
+        ----------
+        entity: DictSerializable or List[DictSerializable]
+            Object to represent using its name. Generally a (list of) BaseEntity or BaseAttribute,
+            both of which have a `name` field.
+
+        Returns
+        -------
+        str
+            A representation of `entity` using its name.
+
+        """
+        if isinstance(entity, (list, tuple)):
+            return [self._name_repr(item) for item in entity]
+        elif entity is None:
+            return None
+        else:
```

Comment:
I think it would be a good idea to just fall back on `repr(entity)` if `entity` isn't actually an instance of `DictSerializable`. I know that won't come up right now but it seems possible that in the future we may want to put a logger or some other unserializable object in `_skip` which violates the assumption about type in this function.
Alternatively, you could do that check in `__repr__` itself.

Reply:
I had something like that initially, but removed it because there was no non-artificial way to test it. There's no way to make `__repr__` call `_name_repr` on an object that isn't a `DictSerializable`, because it only gets called for soft links and soft links cannot be set directly. Therefore in order to get coverage on that line I would have had to make a new test that created a non-DictSerializable object and called `_name_repr` on it, a displeasingly artificial situation.
Note that there are actually no type assumptions here. If we pass an object of type `Logger` that has no `name` field, the representation would be `"<Logger '<unknown name>'>"`. It is easy to in the future modify this if-else block to take other options into account, but since we want 100% coverage I did not want to preemptively check for situations that are presently incapable of occurring.

Reply:
Oh I see, yeah I'm not worried about this at all really then, thanks for explaining!